src/main/java/com/mlaf/hu/proxy/parser/AclXmlParser.java

```diff
@@ -1,195 +1,182 @@
+import jade.domain.FIPAAgentManagement.Envelope;
+import javax.xml.bind.JAXBException;
+import jade.lang.acl.ACLMessage;
+import jade.util.Logger;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.logging.Level;
+import javax.xml.bind.JAXB;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.Marshaller;
+
+/**
+ *
+ * @author Hans
+ */
+public class AclXmlParser {
+
+    private static java.util.logging.Logger XmlParserLogger = Logger.getLogger("XmlParserLogger");
+
+    public static String parseACLToXML(ACLMessage message) {
+        StringWriter marshalledObject = new StringWriter();
+
+        AclObject aclObject = new AclObject();
+        ArrayList<AidObject> receiverList = new ArrayList<>();
+        ArrayList<AidObject> senderList = new ArrayList<>();
+        ArrayList<AidObject> replyToList = new ArrayList<>();
+        AidObject sender = newSender(message);
+        AidObject receiver = newReceiver(message);
+        AidObject replyTo = newReplyTo(message);
+
+        receiverList.add(receiver);
+        senderList.add(sender);
+        replyToList.add(replyTo);
+
+        aclObject.setPerformative("" + getPerformative(message.getPerformative()));
+        aclObject.setSender(senderList);
+        aclObject.setReplyTo(replyToList);
+        aclObject.setReceiver(receiverList);
+        aclObject.setContent(message.getContent());
+        aclObject.setLanguage(message.getLanguage());
+        aclObject.setOntology(message.getOntology());
+        aclObject.setProtocol(message.getProtocol());
+        aclObject.setConversationId(message.getConversationId());
+
+        try {
+            JAXBContext context = JAXBContext.newInstance(AclObject.class);
+            Marshaller marshaller = context.createMarshaller();
+            marshaller.setProperty(Marshaller.JAXB_ENCODING, "UTF-8");
+            marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+            marshaller.setProperty(Marshaller.JAXB_FRAGMENT, Boolean.TRUE);
+            
+            marshaller.marshal(aclObject, marshalledObject);
+        } catch (JAXBException ex) {
+            java.util.logging.Logger.getLogger(AclXmlParser.class.getName()).log(Level.SEVERE, null, ex);
+        }
+        String xml = marshalledObject.toString().replace("&lt;", "<").replace("&gt;", ">");
+        return xml;
+    }
+
+    private static AidObject newReceiver(ACLMessage message) {
+        AidObject receiver = new AidObject();
+        Iterator receiverIterator = message.getAllReceiver();
+        if (receiverIterator.hasNext()) {
+
+            AID aid = (AID) message.getAllReceiver().next();
+
+            ArrayList<String> addressesReceiver = new ArrayList<>();
+            addressesReceiver.addAll(Arrays.asList(aid.getAddressesArray()));
+            receiver.setName(aid.getName());
+            receiver.setAddresses(addressesReceiver);

```

Comment:
Het voelt wat raar. Je set Adresses aan een AID? Moet dit niet een AclObject zijn?

Reply:
Nee. Kijk maar ff in de Jade source code of iets. Address wordt opgeslagen in het AID van de agent.